### PR TITLE
Add MisclassLoss as a generalization of ZeroOneLoss

### PR DESCRIPTION
**Pull request by juliohm**
This PR replaces #119 . I appreciate if you can take a look at it and review. We need more work on this package to make it work with CategoricalArrays.jl I am afraid that maintainers are currently busy with other tasks, and I wonder how I can keep working on improvements.

### CODE REVIEW

File: src/supervised/other.jl [hunk 1]
```diff
@@ -1,3 +1,36 @@
+@doc doc"""
+    MissclassLoss
```
Comment (joshday) [inline src/supervised/other.jl]: typo

**Comment by juliohm** [comment]
I've fixed the typo @joshday as suggested, thanks. Could you please take a second look?

**Comment by joshday** [comment]
I'm gonna wait a bit to let others weigh in if they'd like, especially since multiclass losses haven't been "solved" in LossFunctions and this is a pretty new thing.

**Comment by juliohm** [comment]
Thank you @joshday , I appreciate the time you're putting into it. I agree that at some point we need to better handle multiclass losses. My next PR will be on adding support for categorical values from CategoricalArrays.jl I will see what I can do to at least cover this case.

**Comment by juliohm** [comment]
Hi @joshday, do you think someone else will provide feedback until Friday?

I will start to work locally to add support for categorical arrays assuming this PR is merged.

**Comment by Evizero** [comment]
Hi everyone! I just wanted to drop a quick comment in case anyone is waiting for my input.
First of all, thank you for your interest and effort.

That said, I have nothing to add as I am currently not working on anything julia related and simply dont have the bandwidth available to maintain this package in any way or form. So I would be very happy for Josh to make any judgements regarding this or any other PR here (if he has the time).

best regards

**Comment by joshday** [comment]
Thanks for the note, @Evizero.  I don't have a lot of bandwidth either, so absent any objections, I'll add @juliohm to JuliaML

**Comment by joshday** [comment]
There was a git conflict.  I'll merge this when CI goes green!

**Comment by juliohm** [comment]
Hi @Evizero , thank you for your contributions, it is sad news for us that you are not involved with Julia these days. You've authored so many incredibly useful packages... I hope you are doing well, and I am glad we are always crossing roads in many organisations (JuliaImages, JuliaML, ...).

Thank you @joshday for merging the PR, and for considering an invitation to JuliaML along with @oxinabox, I'd be happy to contribute more actively, and evolve the packages here to address my current research. I am particularly happy with LearnBase.jl and LossFunctions.jl. Other existing initiatives for ML in Julia (excluding deep nets) aren't as nice, and so I think we can revive the work here. It deserves more attention.

**Pull request merged**